# Week 2: Python for Data Analysis
Dataset: SQL_Sales_Dataset_200_Rows.csv (200 orders, 10 columns)

## Q1: Load a CSV file using Pandas and display basic info

In [13]:
import pandas as pd

df = pd.read_csv("SQL_Sales_Dataset_200_Rows.csv")
df.head()

,order_id,customer_name,order_date,category,sub_category,product_name,quantity,unit_price,total_price,region
0,1000,Caleb Davis,2025-01-31,Clothing,Jeans,Jeans Try,2,1197,2394,West
1,1001,Misty Herrera,2025-05-05,Clothing,Jeans,Jeans New,9,1509,13581,South
2,1002,Rick Soto,2025-06-09,Clothing,Shoes,Shoes Us,5,1046,5230,East
3,1003,Sean Haas,2025-02-27,Furniture,Desk,Desk Section,10,292,2920,South
4,1004,Ann Hall,2025-06-01,Furniture,Cabinet,Cabinet Center,2,548,1096,North


In [14]:
print(df.shape)
df.info()

(200, 10)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   order_id       200 non-null    int64 
 1   customer_name  200 non-null    object
 2   order_date     200 non-null    object
 3   category       200 non-null    object
 4   sub_category   200 non-null    object
 5   product_name   200 non-null    object
 6   quantity       200 non-null    int64 
 7   unit_price     200 non-null    int64 
 8   total_price    200 non-null    int64 
 9   region         200 non-null    object
dtypes: int64(4), object(6)
memory usage: 15.8+ KB


In [15]:
df.describe()

,order_id,quantity,unit_price,total_price
count,200.000000,200.000000,200.000000,200.000000
mean,1099.500000,4.910000,2421.495000,12100.535000
std,57.879185,3.017853,1464.016187,10915.548067
min,1000.000000,1.000000,79.000000,79.000000
25%,1049.750000,2.000000,1120.000000,3060.000000
50%,1099.500000,5.000000,2373.500000,8912.000000
75%,1149.250000,7.000000,3541.750000,19350.750000
max,1199.000000,10.000000,4997.000000,47940.000000


## Q2: Handle missing values and duplicates using Pandas

In [16]:
df.isnull().sum()

,0
order_id,0
customer_name,0
order_date,0
category,0
sub_category,0
product_name,0
quantity,0
unit_price,0
total_price,0
region,0


In [17]:
df.duplicated().sum()

np.int64(0)

In [18]:
df = df.dropna()
df = df.drop_duplicates()
df.shape

(200, 10)

No missing values or duplicates were found, so no rows were removed. The dataset has 200 rows and 10 columns.

## Q3: Group data by category and find total revenue

In [19]:
revenue_by_category = df.groupby("category")["total_price"].sum().sort_values(ascending=False)
revenue_by_category

,total_price
category,
Furniture,714399
Grocery,672147
Electronics,549302
Clothing,484259


Furniture generates the highest total revenue, followed by Grocery, Electronics and Clothing.

## Q4: Sort data by multiple columns using Python

In [20]:
df_sorted = df.sort_values(by=["category", "total_price"], ascending=[True, False])
df_sorted.head(10)

,order_id,customer_name,order_date,category,sub_category,product_name,quantity,unit_price,total_price,region
168,1168,Megan Charles,2025-03-26,Clothing,Jeans,Jeans Card,10,4462,44620,West
59,1059,Miss Rebecca Levine,2025-02-28,Clothing,Shirt,Shirt Look,6,4794,28764,East
157,1157,Patricia Gill,2025-03-23,Clothing,Jacket,Jacket Song,7,3915,27405,East
180,1180,Hailey Cobb,2025-02-19,Clothing,Jacket,Jacket Heart,6,4285,25710,East
82,1082,Kenneth Lee,2025-05-06,Clothing,Jacket,Jacket Realize,7,3486,24402,West
42,1042,David Norris,2025-06-10,Clothing,Shoes,Shoes Far,7,3445,24115,South
171,1171,David Dominguez,2025-03-06,Clothing,Shoes,Shoes Each,8,2902,23216,South
128,1128,Gregory Miller,2025-02-21,Clothing,Jacket,Jacket Image,8,2822,22576,South
62,1062,Jessica Hernandez,2025-05-21,Clothing,Shoes,Shoes Care,7,3208,22456,East
99,1099,Amanda Reeves,2025-04-21,Clothing,Jeans,Jeans Family,5,4419,22095,South


The data is sorted by category (A to Z) and, within each category, by total_price from highest to lowest.

## Q5: Create a correlation matrix for numerical columns

In [21]:
corr = df[["quantity", "unit_price", "total_price"]].corr()
corr

,quantity,unit_price,total_price
quantity,1.000000,0.047996,0.655212
unit_price,0.047996,1.000000,0.671318
total_price,0.655212,0.671318,1.000000


total_price is positively correlated with both quantity (about 0.66) and unit_price (about 0.67). quantity and unit_price are almost uncorrelated.